tests/rptest/tests/memory_stress_test.py

```diff
@@ -0,0 +1,130 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+from enum import Enum
+
+from ducktape.mark import ok_to_fail
+from rptest.clients.types import TopicSpec
+from rptest.services.cluster import cluster
+from rptest.services.kaf_consumer import KafConsumer
+from rptest.services.redpanda import ResourceSettings
+from rptest.services.rpk_consumer import RpkConsumer
+from rptest.services.rpk_producer import RpkProducer
+from rptest.services.verifiable_consumer import VerifiableConsumer
+from rptest.tests.redpanda_test import RedpandaTest
+from rptest.utils.mode_checks import skip_debug_mode
+
+
+class Consumers(Enum):
+    RPK = 1
+    KAF = 2
+    VERIFIABLE = 3
+
+
+class MemoryStressTest(RedpandaTest):
+    """
+    Try various ways to reach out-of-memory condition in a broker via Kafka API
+    """
+    def setUp(self):
+        # Override parent setUp so that we don't start redpanda until each test,
+        # enabling each test case to customize its ResourceSettings
+        pass
+
+    @ok_to_fail  # until the fix is delivered
+    @cluster(num_nodes=5)
+    @skip_debug_mode
+    def test_fetch_with_many_partitions(self):
+        """
+        Exhaust memory by consuming from too many partitions in a single Fetch
+        API request.
+        """
+        # memory_mb does not work with debug redpanda build, therefore the test
+        # only makes sense with release redpanda, hence @skip_debug_mode
+        self.redpanda.set_resource_settings(
+            ResourceSettings(memory_mb=256, num_cpus=1))
```

Comment:
cool

Reply:
Actually I'm facing issues running the fixed code with these constraints, so memory will most probably go up to half gig.